clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerNetworkClient.java

```diff
@@ -132,10 +132,18 @@ public Node leastLoadedNode() {
      * Block until the metadata has been refreshed.
      */
     public void awaitMetadataUpdate() {
+        awaitMetadataUpdate(Long.MAX_VALUE);
+    }
+
+    /**
+     * Block waiting on the metadata refresh with a timeout.
+     */
+    public boolean awaitMetadataUpdate(long timeout) {
```

Comment:
The loop below doesn't break when the timeout is reached, right? It will still block indefinitely until the metadata has been refreshed.